server/server.go

```diff
@@ -985,11 +977,11 @@ func getToken(md metadata.MD) string {
 	}
 
 	for _, t := range tokens {
```

Comment:
Can we remove tokens list and just return token as soon as it is extracted from grpc metadata:

line 973:
```golang
token, err := httputil.JoinCookies(common.AuthCookieName, request.Cookies())
if token != "" && err == nil {
	return token
}
```

I hope I'm not missing anything.

Reply:
Yes, that is how I implemented earlier. But then what is the usecase of token list? https://github.com/argoproj/argo-cd/pull/5774/files#diff-78f42ba40d0f10b08c73b7e6bb8376f398e249c963cf549e89591d6b6826b9a4L972.

Reply:
To add one more point, https://github.com/argoproj/argo-cd/pull/5774/files#diff-78f42ba40d0f10b08c73b7e6bb8376f398e249c963cf549e89591d6b6826b9a4L988-L990 was removing invalid tokens (token with fragments != 3 when split by '.'). That is why I added it into to list and than ran through for loop to check token validity.

Reply:
Found PR that introduced `token` list: https://github.com/argoproj/argo-cd/pull/2262/files  Not 100% sure I correctly understand that code but I guess we are trying to get the first valid token (token has 3 fragments ). The Bearer token has a higher priority.


In this the logic you've implemented is correct. Sorry for confustion